src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -453,6 +458,9 @@ private static void DeserializeFrameworkTypeValue(CodeWriter writer, CodeWriterD
         {
             bool includeFormat = false;
 
+            if (frameworkType == typeof(ETag))
+                writer.AppendRaw("new Azure.ETag(");
```

Comment:
Use `writer.Append($"{typeof(ETag)}");` for types. CodeWriter would format the type correctly.

Reply:
Lets put the entire expression here along with `$"{element}.")` part. It would make things a bit less confusing